packages/ec-core/src/operator/selector/tournament.rs

```diff
@@ -1,18 +1,31 @@
+use std::num::NonZeroUsize;
+
 use anyhow::{ensure, Context, Result};
 use rand::{prelude::IndexedRandom, rngs::ThreadRng};
 
 use super::Selector;
 use crate::population::Population;
 
 pub struct Tournament {
-    size: usize,
+    size: NonZeroUsize,
 }
 
 impl Tournament {
+    /// Construct a tournament selector with the given size. This
+    /// will select `size` individuals from the population, randomly
+    /// without replacement, and return the "best" from that set.
     #[must_use]
-    pub const fn new(size: usize) -> Self {
+    pub const fn new(size: NonZeroUsize) -> Self {
         Self { size }
     }
+
```

Comment:
How about we add a `::of_size<N>()` constructor like so:
```suggestion
#[must_use]
pub const fn  of_size<const N: usize>() -> Self {
   Self::new(const { match NonZeroUsize::new(N) {
       Some(x) => x,
       None => panic!("only positive tournament sizes are permitted");
   } }))    
}
```

This would enable us to drop `binary()` if we wish to and construct arbitrary sizes of tournaments with const code, but not from dynamic values like variables. Since the `panic!()` is in a `const` block it only gets executed at compile time and as such all errors that could happen will occur at compile time.